torch/quantization/fx/quantization_patterns.py

```diff
@@ -284,7 +285,20 @@ def __init__(self, quantizer, node):
             self.linear = quantizer.modules[self.linear_node.target]
 
     def convert(self, quantizer, node, load_arg, debug=False, convert_custom_config_dict=None):
+        # Supported combinations are:
+        # quant_type | activation (compute_type) | weight
+        #  static       quint8                      qint8
+        #  dynamic      float32 (quint8)            qint8
+        #  weight_only  float32                    float16
+        supported_dtypes = [
```

Comment:
Do we need to think about supported dtypes for all ops? There are two ways to think about this

1. User is responsible for using the correct qconfig based on the operator type. If an operator gets an unsupported dtype it will throw an error and user should update to the right qconfig.
2. Quantization code will automatically decide if it can/cannot quantize an operator based on the dtype provided by user in the qconfig.

If we want to do 1 then this shouldn't be necessary and we will just fail in the `from_float` call to linear. For 2, I think it gets more complex as we will need this logic for all ops.

Reply:
I think user is responsible for providing correct qconfig, but we can also provide good error message as well if needed.
not sure what you mean by (2), are you talking about a separate function that given a qconfig and the model and produce some message about whether the qconfig makes sense for the input model?

In this PR linear should not be quantized under the float_qparams_dynamic_qconfig setting so we'll need to check for that.

Reply:
> If we want to do 1 then this shouldn't be necessary and we will just fail in the from_float call to linear. For 2, I think it gets more complex as we will need this logic for all ops.

don't we want linear to work with float_qparam_dynamic_qconfig setting? this pr is in the same spirit as the previous PR.

Reply:
> don't we want linear to work with float_qparam_dynamic_qconfig setting?

In this PR we are making sure linear is not quantized for `float_qparam_dynamic_qconfig`, right? I think if we provide good error message for this that would be enough (based on your previous comment)

Reply:
I see, yeah that makes sense, I think we can have some error message and ask users to change their qconfig accordingly. will do this in the previous PR as well.

Reply:
Sounds good. Can we add similar error message for embeddings as well? In that case we expect activation dtype to be fp32 and weight dtype to be quint8 or quint4x2 currently. I can do it in a separate PR if required.

Reply:
yeah, already had that pr here: https://github.com/pytorch/pytorch/pull/47092, will update that with the check for dtype after this one lands.